flent/runners.py

```diff
@@ -1022,12 +1023,19 @@ def check(self):
                 netperf['-e'] = True
 
             try:
-                # Sanity check; is /dev/urandom readable? If so, use it to
+                # If --test-payload option is specified, use data from that file
+                # else use the default value /dev/urandom. 
+                fill_file = args['test_payload']
+                # Sanity check; is /dev/urandom or the custom file readable? If so, use it to
                 # pre-fill netperf's buffers
-                self.run_simple(['dd', 'if=/dev/urandom', 'of=/dev/null', 'bs=1', 'count=1'], errmsg="Err")
-                netperf['buffer'] = '-F /dev/urandom'
+                self.run_simple(['dd', 'if='+fill_file, 'of=/dev/null', 'bs=1', 'count=1'], errmsg="Err")
+                netperf['buffer'] = '-F '+fill_file
             except RunnerCheckError:
-                netperf['buffer'] = ''
+                if(fill_file == '/dev/urandom'):
+                    netperf['buffer'] = ''
+                else:
+                    # If the custom file is not readable, fail noisily
+                    raise RunnerCheckError("The specified fill file does not exist or is not readable.")
```

Comment:
The option is --test-payload, so referring to "fill file" in the error message is going to confuse people. Better stick to "test payload" here as well :)